arrus/core/devices/us4r/Us4RImpl.cpp

```diff
@@ -69,25 +69,34 @@ std::vector<std::pair <std::string,float>> Us4RImpl::logVoltages(bool isUS4PSC)
         voltages.push_back(temp);
     }
 
-    //Verify measured voltages on OEMs
-    for (uint8_t i = 0; i < getNumberOfUs4OEMs(); i++) {
-        voltage = this->getUCDMeasuredHVPVoltage(i);
-        temp = std::make_pair(std::string("HVP on OEM#" + std::to_string(i)), voltage);
-        voltages.push_back(temp);
-        voltage = this->getUCDMeasuredHVMVoltage(i);
-        temp = std::make_pair(std::string("HVM on OEM#" + std::to_string(i)), voltage);
-        voltages.push_back(temp);
+    auto ver = us4oems[0]->getOemVersion();
+
+    if(ver == 1) {
+        //Verify measured voltages on OEMs
+        for (uint8_t i = 0; i < getNumberOfUs4OEMs(); i++) {
+            voltage = this->getUCDMeasuredHVPVoltage(i);
+            temp = std::make_pair(std::string("HVP on OEM#" + std::to_string(i)), voltage);
+            voltages.push_back(temp);
+            voltage = this->getUCDMeasuredHVMVoltage(i);
+            temp = std::make_pair(std::string("HVM on OEM#" + std::to_string(i)), voltage);
+            voltages.push_back(temp);
+        }
+    }
+    else if(ver == 2) {
+        //Verify measured voltages on OEM+s
```

Comment:
Probably some comment why this branch is empty might be helpful for the future.

Reply:
Added comment with clarification